keps/sig-apps/20190523-support-completions-index-for-job.md

```diff
@@ -0,0 +1,78 @@
+---
+title: support-completions-index-for-job
+authors:
+  - "@fudali"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-job
+reviewers:
+  - "@kow3ns"
+  - "@janetkuo"
+approvers:
+  - "@kow3ns"
+  - "@janetkuo"
+editor: TBD
+creation-date: 2019-05-23
+last-updated: 2019-05-27
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# support completions index for JOB
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+
+## Summary
+
+[Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)
+is a Kubernetes CRD that help us run task, Job allows a user to specify the task or parallel task through either `podTemplate` and `completions`; but i use parallel task i must ensure task unrepeat in myself application, this is diffcult; if job support sahrds, i can use shards index to implement task unrepeat is so easy;
+
+## Motivation
+
+I have a case need shards to job, i want to use kubernetes implement my case; i read the [document](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#parallel-jobs), this is a not implement function, so i want implement it;
```

Comment:
hm... it's better to have a general motivation, e.g. resolve a problem that raised by several dev/user.

Reply:
ok, i will addition it;

my motivation is i think shared index is a general feature in job platform, i want to easy to use job when user have relate demand and   maintenance platform less